### PR TITLE
Fix get_banner_image_url error on public domain in some views

### PR DESCRIPTION
**Pull request by bashu**
### What?

Apply this patch to make sure every `/accounts/*` views are not accessible for public tenant schemas, ie. return 404 (not found) for general public.

### Why?

Because of bug #1214 

### How?

Apply `non_public_only_view` decorator on every `allauth` views, also using `NonPublicOnlyViewMixin` for previously overriden views.

Updated / new dependency: `django-decorator-include` library.

`decorator_include` is intended for use in URL confs as a replacement for the `django.conf.urls.include` function. It works in almost the same way as include however the first argument should be either a decorator or an iterable of decorators to apply to all included views (if an iterable, the order of the decorators is the order in which the functions will be applied on the views).

### Testing?
Yes

### Screenshots (if front end is affected)
No

### Anything Else?
No

### Review request
@tylerecouture

### CODE REVIEW

**Comment by tylerecouture** [comment]
image

This just restates the problem.  Please describe how you solve it.

**Comment by bashu** [comment]
> You need to better document what/how you solved the problem in your PR notes. You include a new dependancy. What's it doing?

fixed

**Comment by bashu** [comment]
>  All tests need better docstrings that are actually useful and include What and Why. If it's all the same, them put them all in a single test (or two tests, one for public, one for not)

fixed

File: src/hackerspace_online/tests/test_auth.py [hunk 1]
```diff
@@ -1,17 +1,71 @@
...
 
+class NonPublicOnlyAuthViewsTest(ViewTestUtilsMixin, TenantTestCase):
+    """
+    Custom `non_public_only_view` decorator was applied on every `allauth` views.
+    """
+
+    def setUp(self):
+        # Every test needs access to the request factory.
+        # https://docs.djangoproject.com/en/3.0/topics/testing/advanced/#the-request-factory
+        # self.factory = RequestFactory()
```
Comment (tylerecouture) [inline src/hackerspace_online/tests/test_auth.py:25]: I don't understand the purpose of these comments here.

**Review by tylerecouture — CHANGES_REQUESTED** [review]
`test_password_reset_view` was deleted and you didn't address the review comment from the last review.

Please address every comment and mark as resolved when you think it has been resolved.

**Comment by tylerecouture** [comment]
> > You need to better document what/how you solved the problem in your PR notes. You include a new dependancy. What's it doing?
> 
> fixed

You should mark the comment as resolved and put this as a reply to the comment.  This makes reviews easier. Thank you!

File: src/hackerspace_online/tests/test_views.py [hunk 2]
```diff
@@ -88,9 +88,6 @@ def test_favicon_public_tenant(self, mock_connection1, mock_connection2):
         self.assertEqual(response.status_code, 301)  # permanent redirect
         self.assertEqual(response.url, static('icon/favicon.ico'))
 
-    def test_password_reset_view(self):
```
[the remark this answers is not in the document]
Comment (bashu) [reply src/hackerspace_online/tests/test_views.py:91]: to @tylerecouture I am sorry, I missed this comment. All `test_password_reset_view`  do, is test whether `account_reset_password` view returns 200 (ok) for non-public tenant or not. I moved it into one single test case (see, test_auth.py)

File: src/hackerspace_online/tests/test_views.py [hunk 3]
```diff
@@ -104,6 +101,224 @@ def test_achievements_redirect_to_badges_views(self):
...
+
+    def setUp(self):
+        # Every test needs access to the request factory.
+        # https://docs.djangoproject.com/en/3.0/topics/testing/advanced/#the-request-factory
+        # self.factory = RequestFactory()
+        self.client = TenantClient(self.tenant)
+
+    @patch('hackerspace_online.views.connection', schema_name=get_public_schema_name())
+    @patch('tenant.views.connection', schema_name=get_public_schema_name())
+    def test_sigunp_public_tenant(self, mock_connection1, mock_connection2):
```
[the remark this answers is not in the document]
Comment (bashu) [reply src/hackerspace_online/tests/test_views.py]: fixed

File: src/hackerspace_online/tests/test_auth.py (hunk 1 above)
Comment (bashu) [reply src/hackerspace_online/tests/test_auth.py:25]: bad copy-paste

File: src/hackerspace_online/tests/test_views.py [hunk 4]
```diff
@@ -104,6 +101,224 @@ def test_achievements_redirect_to_badges_views(self):
         self.assertRedirects(self.client.get('/achievements/1/delete/'), reverse('badges:badge_delete', args=[1]))
 
 
+class AccountOverridenVewsTest(ViewTestUtilsMixin, TenantTestCase):
+    """Misc. tests for overriden (decorated) `allauth` views"""
```
[the remark this answers is not in the document]
Comment (bashu) [reply src/hackerspace_online/tests/test_views.py]: already fixed

File: src/hackerspace_online/tests/test_views.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (tylerecouture) [reply src/hackerspace_online/tests/test_views.py:91]: Makes sense!  Thank you.

**Pull request merged**